File: java/src/test/java/challenges/aoc2024/day2/model/ReactorReportTest.java
package challenges.aoc2024.day2.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class ReactorReportTest {
    
    @Test
    void shouldBuildReactorReportObjectFromString() {
        // Given
        final String input = "7 6 4 2 1";

        // When
        final ReactorReport reactorReport = ReactorReport.of(input);

        // Then
        assertEquals(ImmutableReactorReport.class, reactorReport.getClass());
        assertEquals(5, reactorReport.getLevels().size());
        assertEquals(7, reactorReport.getLevels().get(0));
        assertEquals(6, reactorReport.getLevels().get(1));
        assertEquals(4, reactorReport.getLevels().get(2));
        assertEquals(2, reactorReport.getLevels().get(3));
        assertEquals(1, reactorReport.getLevels().get(4));
    }
}

File: scala/src/test/scala/challenges/aoc2024/day2/model/ReactorReportTest.scala
package challenges.aoc2024.day2.model

import org.scalatest.funsuite.AnyFunSuite

class ReactorReportTest extends AnyFunSuite {

  test("should build ReactorReport object from string") {
    // Given
    val input: String = "7 6 4 2 1"

    // When
    val reactorReport: ReactorReport = ReactorReport.of(input)

    // Then
    assert(reactorReport.isInstanceOf[ReactorReport])
    assert(5 == reactorReport.levels.size)
    assert(7 == reactorReport.levels.head)
    assert(6 == reactorReport.levels(1))
    assert(4 == reactorReport.levels(2))
    assert(2 == reactorReport.levels(3))
    assert(1 == reactorReport.levels(4))
  }
}
